src/main.rs

```diff
@@ -47,6 +53,7 @@ async fn async_main() -> BoxResult<()> {
     database.run_command(doc!("ping": 1), None).await?;
     println!("Connected to mongo database.");
     let bluetooth_frames = database.collection::<BluetoothFrame>("bluetooth_frames");
+    let heartbeat_frames = database.collection::<HeartbeatFrame>("heartbeat_frames");
```

Comment:
This would be a good use case for enums. The serde library provides tools to select the right enum parser and I think there is one that would solve this. 

Despite that I think this is a 10/10 commit.